btree/btree.c

```diff
@@ -6808,6 +6902,43 @@ static void btree_ut_kv_oper_thread_handler(struct btree_ut_thread_info *ti)
 			keys_put_count++;
 			key_first += ti->ti_key_incr;
 		}
+
+		/** UPDATE record which we inserted above. */
+
+		key_first = key_iter_start;
+		ut_cb.c_act   = btree_kv_update_cb;
+		ut_cb.c_datum = &data;
+
+		while (key_first <= key_last) {
+			/**
+			 *  Embed the thread-id in LSB so that different threads
+			 *  will target the same node thus causing race
+			 *  conditions useful to mimic and test btree operations
+			 *  in a loaded system.
+			 */
+			key[0] = (key_first << (sizeof(ti->ti_thread_id) * 8)) +
+				 ti->ti_thread_id;
+			key[0] = m0_byteorder_cpu_to_be64(key[0]);
+			for (i = 1; i < ARRAY_SIZE(key); i++)
+				key[i] = key[0];
+
+			value[0] = key[0];
+			for (i = 1; i < ARRAY_SIZE(value); i++)
```

Comment:
keeping value same as key, as btree_kv_get_cb will fail if value is different than key